unit_tests/src/filesystem/test_filesystem.cpp

```diff
@@ -180,6 +264,81 @@ TEST_CASE("filesystem")
         REQUIRE(fs::client::remove(conn, col, fs::remove_options::no_trash));
     }
 
+    SECTION("fetch data object mtime")
+    {
+        using namespace std::chrono_literals;
+
+        const fs::path p = sandbox / "data_object";
+
+        {
+            irods::experimental::client_connection conn;
+            default_transport tp{conn};
+            odstream{tp, p} << "hello world!";
+        }
+
+        const auto first_replica_mtime = fs::client::last_write_time(conn, p);
+        std::this_thread::sleep_for(2s);
+
+        namespace adm = irods::experimental::administration;
+
+        // Get hostname so that we can create a new resource.
+        char host_name[64] {};
+        REQUIRE(gethostname(host_name, sizeof(host_name)) == 0);
+
+        // The new resource's information.
+        adm::resource_registration_info ufs_info;
+        ufs_info.resource_name = "unit_test_ufs";
+        ufs_info.resource_type = adm::resource_type::unixfilesystem.data();
+        ufs_info.host_name = host_name;
+        ufs_info.vault_path = "/tmp";
+
+        irods::at_scope_exit remove_resources{[&ufs_info] {
+            irods::experimental::client_connection conn;
+            adm::client::remove_resource(conn, ufs_info.resource_name);
+        }};
+
+        {
+            // Create a new resource.
+            irods::experimental::client_connection conn;
+            REQUIRE(adm::client::add_resource(conn, ufs_info).value() == 0);
+        }
+
+        {
+            // Replicate the existing data object and verify that the time
+            // returned comes from the most up-to-date good replica.
```

Comment:
most 'recent'.... they're all up-to-date?

Reply:
yes, but i think the mtimes are different so the mtime of the second replica will be returned.